### PR TITLE
[WebProfilerBundle] Profiler View Latest should preserve all the current query parameters

### PR DESCRIPTION
**Pull request by jbafford**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

This improves the View Latest link by preserving all of the current query parameters, which makes it more useful with the Logs panel, for example.

### CODE REVIEW

File: src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php [hunk 1]
```diff
@@ -112,6 +112,7 @@ public function panelAction(Request $request, $token)
             'request' => $request,
             'templates' => $this->getTemplateManager()->getTemplates($profile),
             'is_ajax' => $request->isXmlHttpRequest(),
+            'all_query' => $request->query->all(),
```
Comment (stof) [inline src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php]: the Request is available in the template, so passing a new variable is not needed

File: src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/layout.html.twig [hunk 2]
```diff
@@ -14,7 +14,7 @@
                     {% if profile %}
                         <div id="resume">
                             <a id="resume-view-all" href="{{ path('_profiler_search', {limit: 10}) }}">View last 10</a>
-                            <a id="resume-view-latest" href="{{ path('_profiler', {token: 'latest', 'panel': panel}) }}">View latest</a>
+                            <a id="resume-view-latest" href="{{ path('_profiler', {token: 'latest'} + all_query) }}">View latest</a>
```
Comment (stof) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/layout.html.twig]: you should use the ``|merge()`` filter. I don't think using the ``+`` to merge arrays is a supported use case in Twig (it is not a documented one for sure)

**Comment by fabpot** [comment]
:+1:

**Comment by stof** [comment]
This needs to be rebased on top of the new profiler.

but otherwise, :+1: 

status: reviewed

**Comment by jbafford** [comment]
It appears the View Latest button got removed altogether at some point with the new profiler. This rebase re-adds it.

Because there's limited space to fit the buttons, I've made the following changes:
* s/Latest profiles/Last 10/
* s/View Latest/Latest/
* Changed the left/right padding on the buttons from .75em to .5em

These changes allow the buttons to fit on one line, since it looks pretty bad if they're split on multiple lines, but it's still a pretty tight fit.

Unfortunately, other changes to the new profiler break the original problem I was trying to resolve (specifically: reloading the logs page with the current filter still selected), but this is still better than what we had before.

**Comment by javiereguiluz** [comment]
@jbafford the "View Latest" button is displayed in the new profiler. Just tested:

view-latest

**Comment by jbafford** [comment]
@javiereguiluz In an earlier 2.8 commit, there was actually two buttons: one to view the list of the last ten, but also a button that was added that showed the current profiler page on the latest request (so that you didn't have to click last 10, click the most recent, and then click back to your current profiler page). The second button got removed at some point when the new profiler UI was added.

**Comment by javiereguiluz** [comment]
@jbafford I think we should still get rid of this button ... and add it as a link in the "Symfony Profiler" page heading. "Going back to home" is the behaviour that people expect when clicking on the website heading.

**Comment by fabpot** [comment]
We need a decision here. I trust @javiereguiluz about these kind of things, but if anyone want to give their 2c, please do so now.

**Comment by javiereguiluz** [comment]
After rethinking about this, I think I was wrong. Putting this link in the `<h1>` element does not make sense because a request can have sub-request and then the navigation flow is not correct.

@jbafford please do the following changes in `src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/profiler.css.twig` to be able to merge this:

1) Revert the change that you made on the `padding` value of the `. btn` selector. It should be:

```css
.btn {
    padding: .5em .75em;
}
```

2) Change the `margin-left` value (which was `10px`) in this selector:

```css
#sidebar #sidebar-shortcuts .btn + .btn {
    margin-left: 5px;
}
```

3) Create this new rule to change the pading only for the shortcut buttons, not to the other buttons:

```css
#sidebar #sidebar-shortcuts .btn {
    padding: .5em .6em;
}
```

This is how it should look:

profiler-buttons

**Comment by jbafford** [comment]
@javiereguiluz I've rebased against the current 2.8 and made the requested changes.

**Comment by javiereguiluz** [comment]
:+1: thanks @jbafford

**Comment by fabpot** [comment]
Thank you @jbafford.

**Pull request merged**